v2/src/features/UploadList/UploadList.tsx

```diff
@@ -52,11 +52,28 @@ const generateErrorPayload = (error: AxiosError | Error) => {
   return errorPayload;
 };
 
+const LOCAL_STORAGE_KEY = 'sugarshare.files';
+
+
+const getOnlocalStorage = (key: string) => {
+  const result = window.localStorage.getItem(key) || '[]';
+  return JSON.parse(result);
+};
+
+const INIT_STATE = getOnlocalStorage(LOCAL_STORAGE_KEY)
+  ? (getOnlocalStorage(LOCAL_STORAGE_KEY) as SugarFileState[])
```

Comment:
Since there is no caching, this is making 2 reads from the disk?

Reply:
yes

Reply:
Soooo you gonna fix it right?

Reply:
updated